x-pack/plugins/ml/public/ui_actions/open_in_anomaly_explorer_action.tsx

```diff
@@ -25,42 +34,96 @@ export function createOpenInExplorerAction(getStartServices: MlCoreSetup['getSta
         defaultMessage: 'Open in Anomaly Explorer',
       });
     },
-    async getHref({ embeddable, data }): Promise<string> {
+    async getHref(context): Promise<string | undefined> {
       const [, pluginsStart] = await getStartServices();
       const urlGenerator = pluginsStart.share.urlGenerators.getUrlGenerator(ML_APP_URL_GENERATOR);
-      const { jobIds, timeRange, viewBy } = embeddable.getInput();
-      const { perPage, fromPage } = embeddable.getOutput();
 
-      return urlGenerator.createUrl({
-        page: 'explorer',
-        pageState: {
-          jobIds,
-          timeRange,
-          mlExplorerSwimlane: {
-            viewByFromPage: fromPage,
-            viewByPerPage: perPage,
-            viewByFieldName: viewBy,
-            ...(data
-              ? {
-                  selectedType: data.type,
-                  selectedTimes: data.times,
-                  selectedLanes: data.lanes,
-                }
-              : {}),
+      if (isSwimLaneEmbeddable(context)) {
+        const { embeddable, data } = context;
+
+        const { jobIds, timeRange, viewBy } = embeddable.getInput();
+        const { perPage, fromPage } = embeddable.getOutput();
+
+        return urlGenerator.createUrl({
+          page: 'explorer',
+          pageState: {
+            jobIds,
+            timeRange,
+            mlExplorerSwimlane: {
+              viewByFromPage: fromPage,
+              viewByPerPage: perPage,
+              viewByFieldName: viewBy,
+              ...(data
+                ? {
+                    selectedType: data.type,
+                    selectedTimes: data.times,
+                    selectedLanes: data.lanes,
+                  }
+                : {}),
+            },
           },
-        },
-      });
+        });
+      } else if (isAnomalyExplorerEmbeddable(context)) {
+        const { embeddable } = context;
+
+        const { jobIds, timeRange } = embeddable.getInput();
+        const { entityFields } = embeddable.getOutput();
+
+        let mlExplorerFilter: ExplorerAppState['mlExplorerFilter'] | undefined;
+        if (
+          Array.isArray(entityFields) &&
+          entityFields.length === 1 &&
+          entityFields[0].operation === ENTITFY_FIELD_OPERATIONS.ADD
+        ) {
+          const { fieldName, fieldValue } = entityFields[0];
+          if (typeof fieldName === 'string' && typeof fieldValue === 'string') {
```

Comment:
question: does a field value always have to be a string?

Reply:
Updated here https://github.com/elastic/kibana/pull/94396/commits/5db8eb1637e3ee18c36b5db73aa06a10b1acd339. I think I originally added this check here because the type of `mlExplorerFilter.filteredFields` is `string[]`. Should I change it to include other types as well or cast fieldValue to be a string?